### PR TITLE
Add Non-functional requirements

### LINKED ISSUE #305: Update Appendix D: Non-Functional Requirements of DG

### PR DESCRIPTION
**Pull request by UncleGrandpa925**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -621,13 +621,20 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 ## **Appendix D: Non-Functional Requirements** <a name="appen-d"></a>
 
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 tasks/contacts/classes without a noticeable sluggishness in performance for
- typical usage.
-3.  Users of the app find the app intuitive and easy to use
-4.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should
+2.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should
```
Comment (simonteozw) [inline docs/DeveloperGuide.md]: Think we can bold the `above average typing speed`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -621,13 +621,20 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-*{More to be added}*
+3.  A user should be able to easily see why their commands are invalid
+4.  The app should be able to run with or without internet connection
+5.  The app should not require user to install
+6.  Features implemented should be easily testable by manual testing, and possisble to be tested by automated testing.
+7.  The UI of the app makes users comfortable using it
+8.  Users of the app find the app intuitive and easy to use
+9.  The app should be able to save data locally
+10. The app should be for a single user i.e. (not a multi-user product).
+11. The app should be able to hold up to 150 modules/lessons/tasks/contacts without a noticeable sluggishness in performance for
```
Comment (simonteozw) [inline docs/DeveloperGuide.md]: 150 means 150 of each or 150 in total?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -621,13 +621,20 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+ typical usage.
+12. The app should be able to start up sufficiently fast (<=5s).
```
Comment (simonteozw) [inline docs/DeveloperGuide.md]: I think we just write <5s, <=5 looks ugly

**Review by simonteozw — COMMENTED** [review]
Looks good! Just some comments

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (UncleGrandpa925) [reply docs/DeveloperGuide.md]: oh yeah!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (UncleGrandpa925) [reply docs/DeveloperGuide.md]: in total, lemme clarify it

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (UncleGrandpa925) [reply docs/DeveloperGuide.md]: Sure!

Comment (UncleGrandpa925) [reply docs/DeveloperGuide.md]: Done!

**Review by simonteozw — APPROVED** [review]
LGTM

**Pull request merged**